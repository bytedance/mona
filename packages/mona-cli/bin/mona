#! /usr/bin/env node

const mona = require('../dist').default;
mona();
