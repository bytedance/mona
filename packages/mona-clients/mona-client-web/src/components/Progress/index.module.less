.progress {

  & .inner {
    width: 0;
    height: 100%;

    &.active {
      transition-property: width;
      transition-duration: 1s;
      transition-timing-function: linear;
    }
  }
}