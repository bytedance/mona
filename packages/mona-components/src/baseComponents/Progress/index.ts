import React from 'react';
import createComponent from '../../createComponent';
import { BaseProps } from '../base';

export interface ProgressProps extends BaseProps {
  percent?: number;
  strokeWidth?: number;
  color?: string;
  activeColor?: string;
  backgroundColor?: string;
  active?: boolean;
  activeMode?: string;
}

export const Progress: React.ComponentType<ProgressProps> = createComponent<ProgressProps>('progress')
