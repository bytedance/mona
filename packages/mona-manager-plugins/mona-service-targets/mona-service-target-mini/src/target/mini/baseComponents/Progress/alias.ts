import { bubbleEventsAlias, basePropAlias, propAliasMap } from '../prop';
import { ProgressAlias as Alias } from '../type';

const alias: Alias = {
  ...basePropAlias,
  ...bubbleEventsAlias,
  percent: propAliasMap.percent,
  strokeWidth: propAliasMap.strokeWidth,
  color: propAliasMap.color,
  activeColor: propAliasMap.activeColor,
  backgroundColor: propAliasMap.backgroundColor,
  active: propAliasMap.active,
  activeMode: propAliasMap.activeMode,
};

export default alias;
