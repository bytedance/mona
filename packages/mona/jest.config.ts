import baseConfig from '../../jestconfig.base';

module.exports = baseConfig;
