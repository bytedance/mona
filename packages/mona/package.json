{
  "name": "@bytedance/mona",
  "version": "0.2.0-alpha.9",
  "description": "Merchant app's developing and building tools",
  "main": "./dist/index",
  "scripts": {
    "build": "yarn clean && ttsc -b",
    "start": "ttsc -b -w",
    "clean": "rm -rf dist",
    "test": "jest"
  },
  "files": [
    "dist"
  ],
  "author": "zhangxiaowei.chris@bytedance.com",
  "license": "MIT",
  "publishConfig": {
    "access": "public",
    "registry": "https://registry.npmjs.org"
  },
  "devDependencies": {
    "copy-webpack-plugin": "^10.2.0",
    "react": "^17.0.2",
    "webpack": "^5.58.2",
    "webpack-chain": "^6.5.1"
  },
  "peerDependencies": {
    "react": "^17.0.2"
  },
  "gitHead": "cc5aa5283af595fd7d8a7b4921f54e2797504edd"
}
