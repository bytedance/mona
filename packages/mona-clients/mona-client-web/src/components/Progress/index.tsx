import React, { useEffect, useRef, useState } from 'react';
import { ProgressProps } from '@bytedance/mona';
import styles from './index.module.less';
import { useHandlers } from '../hooks';

function format(percent: number) {
  return percent < 0 ? 0 : percent > 100 ? 100 : percent;
}

const Progress: React.FC<ProgressProps> = props => {
  const {
    percent = 0,
    strokeWidth = 6,
    color = '#F85959',
    activeColor = '#F85959',
    backgroundColor = '#ebebeb',
    active = false,
    activeMode = 'backwards',
    style = {},
    ...restProps
  } = props;
  const eleRef = useRef<HTMLDivElement | null>(null);
  const gapRef = useRef(0);
  const prevRef = useRef(0);
  const [p, setP] = useState(0);

  useEffect(() => {
    if (activeMode === 'forwards') {
      gapRef.current = Math.abs(format(percent) - prevRef.current);
      setP(format(percent));
    } else {
      gapRef.current = 0;
      setP(0);
      setTimeout(() => {
        gapRef.current = format(percent);
        setP(format(percent));
      }, 0);
    }
    prevRef.current = format(percent);
  }, [percent]);

  const { handleClassName, ...handleProps } = useHandlers(restProps);
  // const gap = percent - prevRef.current;

  return (
    <div
      className={handleClassName(styles.progress)}
      style={{ ...style, backgroundColor, height: strokeWidth }}
      {...handleProps}
    >
      <div
        ref={eleRef}
        className={`${styles.inner} ${active ? styles.active : ''}`}
        style={{
          width: `${p}%`,
          backgroundColor: activeColor || color,
          transitionDuration: `${gapRef.current * 30}ms`,
        }}
      ></div>
    </div>
  );
};

export default Progress;
