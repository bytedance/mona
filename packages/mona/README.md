# @bytedance/mona
对外暴露提供构建时API
## 配置创建
### createProjectConfig
创建项目配置，`mona.config.ts`中使用
#### 例子
```js
// mona.config.ts
import { createProjectConfig } from '@bytedance/mona';
export default createProjectConfig({
  projectName: 'demo',
  input: './src/app.tsx',
  output: 'dist'
})
```

#### 参数说明
|  参数   | 说明  | 是否必填 | 类型 | 默认值 |
|  ----  | ----  | ---- | ---- | ---- |
| projectName  | 应用/插件名称 | 是 | string | - |
| input  | 入口文件路径 | 否 | string | - |
| output  | 打包后的文件目录路径 | 否 | string | - |
| raw  | 定义函数来修改webpack配置 | 否 | (options: any) => any | - |
| dev  | 用来修改本地服务配置 | 否 | string | { port: 9999 } |

### createAppConfig
创建应用配置，`app.config.ts`中使用
#### 例子
```js
// app.config.ts
import { createAppConfig } from '@bytedance/mona';
export default createAppConfig({
  pages: [
    'pages/Home/index',
    'pages/Info/index',
    'pages/List/index'
  ]
})
```
#### 参数说明
|  参数   | 说明  | 是否必填 | 类型 | 默认值 |
|  ----  | ----  | ---- | ---- | ---- |
| pages  | 页面路径 | 是 | string[] | - |

默认`pages`的第一项为首页
### createPageConfig
创建页面配置，需在每个页面目录的`page.config.ts`中使用

#### 例子
```js
// page.config.ts
import { createPageConfig } from '@bytedance/mona';
export default createPageConfig({
  navigationBarTitleText: 'page title'
})
```
#### 参数说明
|  参数   | 说明  | 是否必填 | 类型 | 默认值 |
|  ----  | ----  | ---- | ---- | ---- |
| navigationBarTitleText  | 当前页面标题 | 否 | string | - |
