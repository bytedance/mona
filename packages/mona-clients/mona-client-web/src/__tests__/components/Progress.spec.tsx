import React from 'react';
import { mount } from 'enzyme';
import Progress from '../../components/Progress';
import mountTest from '../../../../../tests/shared/mountTest';
import handlerTest from '../../../../../tests/shared/handlerTest';
import { ProgressProps } from '@bytedance/mona';

describe('web component: Progress', () => {
  mountTest(Progress)
  handlerTest({ Component: Progress, firstTag: 'div' });

  it('should render correctly', () => {
    expect(mount(<Progress />).render()).toMatchSnapshot();
  })
  
  it('should have right properties', () => {
    const props: ProgressProps = {
      percent: 50,
      strokeWidth: 10,
      color: 'red',
      activeColor: 'blue',
      backgroundColor: 'yellow',
      active: true,
      activeMode: 'forwards',
      className: 'class-name',
      style: { color: 'red' },
    }
    const wrapper = mount(<Progress {...props}>hello</Progress>);
    expect(wrapper.render()).toMatchSnapshot();
  })
})
