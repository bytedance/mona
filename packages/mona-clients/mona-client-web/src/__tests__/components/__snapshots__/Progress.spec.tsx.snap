// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`web component: Progress should have right properties 1`] = `
LoadedCheerio {
  "0": <div
    class="progress class-name  "
    style="color: red; background-color: yellow; height: 10px;"
  >
    <div
      class="inner active"
      style="width: 50%; background-color: blue; transition-duration: 1500ms;"
    />
  </div>,
  "_root": LoadedCheerio {
    "0": Document {
      "children": Array [
        <html>
          <head />
          <body />
        </html>,
      ],
      "endIndex": null,
      "next": null,
      "parent": null,
      "prev": null,
      "startIndex": null,
      "type": "root",
      "x-mode": "quirks",
    },
    "_root": [Circular],
    "length": 1,
    "options": Object {
      "decodeEntities": true,
      "xml": false,
    },
  },
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "xml": false,
  },
}
`;

exports[`web component: Progress should render correctly 1`] = `
LoadedCheerio {
  "0": <div
    class="progress   "
    style="background-color: rgb(235, 235, 235); height: 6px;"
  >
    <div
      class="inner "
      style="width: 0%; background-color: rgb(248, 89, 89); transition-duration: 0ms;"
    />
  </div>,
  "_root": LoadedCheerio {
    "0": Document {
      "children": Array [
        <html>
          <head />
          <body />
        </html>,
      ],
      "endIndex": null,
      "next": null,
      "parent": null,
      "prev": null,
      "startIndex": null,
      "type": "root",
      "x-mode": "quirks",
    },
    "_root": [Circular],
    "length": 1,
    "options": Object {
      "decodeEntities": true,
      "xml": false,
    },
  },
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "xml": false,
  },
}
`;
